lib/components/base/Badge.vue

```diff
@@ -53,16 +53,15 @@ import {
   capitalizeString,
 } from '@'
 
-defineProps({
-  type: {
-    type: String,
-    required: true,
-  },
-  color: {
-    type: String,
-    default: '',
-  },
-})
+withDefaults(
+  defineProps<{
+    type: string
```

Comment:
I think since we're committing to using TypeScript it'd make sense to use a union of type strings here to provide suggestions, but allow free type values using `(string & Record<never, never>)` as part of the union.

Here's a [Vue playground with example](https://play.vuejs.org/#eNqNVctyIzUU/ZU7vaCTKqcTSEEVHsfUQGVqAgVMJUPNgmYhd1/bmqglIan9wPgbWLBjAws+gu/hB+ATOJLfwUlNsrCke+7RfZyrXmQvrC0mLWfdrOcrJ20gz6G1/VLLxhoXaEGOh7SkoTMN5YDmW9MXprHr8+I8biJT/rzUpa6M9oEaP6Kr6H+Sv2KlDL01TtXP8tNS985X1+EibAI3VonA2BH1Em+YW74qs6lwWupRmVFllHE4cVyXWQISvRkzXQsXxrkna5RwJCumSlhPwjE1rAKcSQQSmgTsTdw63ER1yxQMVUo2cVuNhR5xhxSLOmJgctLHlWeB0wkrX6ToUqarQAfu/wFrE3gvWuMi7zbgF2SF81yR9AQkCWq1DGSGFGSD64XHOrCmRvpW1+x8MKbu0KDdx9bSB6ErLugmIHP+sZUToViHGLYYGIAvi4tPKTiplDSa7qUyDQfQkXEkrHVmlvJWcyA/+oSUHI3D2ZwRXCSlKQIZisa0HogGzGDhmqSmgJLfBVT6LbAxoglYmaZjBP0KVb5D4uSFrAnARtSoMhZfsfes6LbVkUNhh2Ogw5TVhNdF8R2CDsbCy59SC+D3NRJgrWXb0EuhoCqk6y1z/V69eCmq0EIKjdGj1u91ZaDa2JPbmBIIW+/nNBD3gm4Q+pTGsqEJki62F/TOtxLNOlnwiGQoR8U7bzRGZxFvLrMKWKnYfWtjuXBfl5Il2gTkP/0ynQXXcmdzXo25uj9y/s7P4lmZvXbs2U0Q7taG6o84rMzXd9/wDOutsTF1q4B+wnjL3qg2xriCfQ6lIew9XIr2Jk05OvHGX8+gSb9JKgYakcuELzNMfSzTY6nvwr0sLpNfqZeo4ubFePj2kMLIoEkBNP3SlRiDodT82hnre4t4EHlji7uUr9+HnH6mPI5eXJx4CB8C+oBuuTKu7mmMr+tQ+umfdjYUSQ2fgQRPSiJYDWtaRoXkCbnsn5zG392LFTe7F2tN1qvlhLqVEt4j9u8Td4fyqIn8h3UiEbfFbqBBhtivA0ACeYsBmZzJYQQhW7q6utolDA9MoG716Bn+6O/f/vjnr18QI5wepcIbxg/5UtFA9l3TkKj8uEXv5/Tv73/++h5k/cUiNYKWyyPo3jnyfDxxzBBUBWEcuULhbTx/gmy325vMtPVhrpgwoJbrLbyIbYAi9/isqOND36UPuXm+bxhAMexwbmeEOZH1EeuZw1ei9QBd2NkD++zMj0Vtpl26wP/He/ZlXByEVEB5h2El3XTxzdxde8RtJdSjnivTU85R2kddo+EJRyqSVA9dh2ji2ZTj5wMERqE4hxTQRexIP1v+B15owtI=).